src/compile/common.ts

```diff
@@ -35,42 +35,61 @@ export function getStyles(mark: MarkDef): string[] {
   return [].concat(mark.type, mark.style ?? []);
 }
 
-export function getMarkPropOrConfig<P extends keyof MarkConfig>(channel: P, mark: MarkDef, config: Config) {
-  return getFirstDefined(mark[channel], getMarkConfig(channel, mark, config));
+export function getMarkPropOrConfig<P extends keyof MarkDef>(
+  channel: P,
+  mark: MarkDef,
+  config: Config,
+  opt: {
+    vgChannel?: VgEncodeChannel;
+    ignoreVgConfig?: boolean;
+  } = {} // Note: Ham: I use `any` here like in getMarkConfig below
+): MarkDef[P] {
+  const {vgChannel, ignoreVgConfig} = opt;
+  if (vgChannel && mark[vgChannel] !== undefined) {
+    return mark[vgChannel];
+  } else if (mark[channel] !== undefined) {
+    return mark[channel];
+  } else if (ignoreVgConfig && (!vgChannel || vgChannel === channel)) {
+    return undefined;
+  }
+
+  return getMarkConfig(channel, mark, config, opt);
 }
 
 /**
  * Return property value from style or mark specific config property if exists.
  * Otherwise, return general mark specific config.
  */
-export function getMarkConfig<P extends keyof MarkConfig>(
+export function getMarkConfig<P extends keyof MarkDef>(
   channel: P,
   mark: MarkDef,
   config: Config,
   {vgChannel}: {vgChannel?: any} = {} // Note: Ham: I use `any` here as it's too hard to make TS knows that MarkConfig[vgChannel] would have the same type as MarkConfig[P]
-): MarkConfig[P] {
-  return getFirstDefined(
+): MarkDef[P] {
+  return getFirstDefined<MarkDef[P]>(
     // style config has highest precedence
     vgChannel ? getMarkStyleConfig(channel, mark, config.style) : undefined,
     getMarkStyleConfig(channel, mark, config.style),
     // then mark-specific config
     vgChannel ? config[mark.type][vgChannel] : undefined,
-    config[mark.type][channel],
+
+    config[mark.type][channel as any], // Need to cast because MarkDef isn't 100% match with AnyMarkConfig, but if the type isn't available, we'll get nothing here, which is fine
```

Comment:
```suggestion
    config[mark.type][channel as any], // Need to cast because MarkDef doesn't 100% match with AnyMarkConfig, but if the type isn't available, we'll get nothing here, which is fine
```